.github/workflows/reusable-python-build_poetry_package.yml

```diff
@@ -0,0 +1,122 @@
+            value: ${{ jobs.build_package.outputs.version }}
+    secrets:
+        JFROG_ARTIFACTORY_URL:
+            description: 'JFrog Artifactory URL'
+            required: true
+        JFROG_ARTIFACTORY_TOKEN:
+            description: 'JFrog Artifactory Token'
+            required: true
+
+defaults:
+    run:
+        shell: 'bash -l {0}'
+
+jobs:
+    build_package:
+        name: Build poetry package
+        runs-on: ${{ inputs.os }}
+        timeout-minutes: ${{ inputs.timeout-minutes }}
+        outputs:
+            build-dir-path: ${{ steps.define-build-path.outputs.dir-path }}
+            version: ${{ steps.get-version.outputs.version }}
+        steps:
+            -   name: Checkout
+                uses: actions/checkout@v4
+                with:
+                    fetch-depth: 0
+            -   name: Get version
+                id: get-version
+                run: |
+                    if [ "${{ inputs.version-tag }}" = "v0.0.0" ]; then
+                        echo "Version tag not available"
+                        exit 1
+                    fi
+                    VERSION=$(echo "${{ inputs.version-tag }}" | sed 's/v//')
+                    echo "version: ${VERSION}"
+                    if [ -z "${VERSION}" ]; then
+                        echo "Version is not available"
+                        exit 1
+                    fi
+                    echo "version=${VERSION}" >> $GITHUB_OUTPUT
+            -   uses: jfrog/setup-jfrog-cli@v4.4.1
+                name: Setup JFrog CLI
+                env:
+                    JF_URL: https://${{ secrets.JFROG_ARTIFACTORY_URL }}
+                    JF_ACCESS_TOKEN: ${{ secrets.JFROG_ARTIFACTORY_TOKEN }}
+        
+            -   name: Set up Python version
+                uses: actions/setup-python@v5
+                with:
+                    python-version: ${{ inputs.python-version }}
+            -   name: Install poetry
+                run: |
+                    curl -sSL https://install.python-poetry.org | POETRY_HOME=$HOME/.poetry python -
+                    echo "$HOME/.poetry/bin" >> $GITHUB_PATH
+            -   name: Configure poetry 
+                run: |
+                    poetry config virtualenvs.create true
+                    poetry config virtualenvs.in-project true
+                    
+                    repository_list=$(echo '${{ inputs.virtual-repo-names }}' | jq -r '.[]')
+                    for repo_name in ${repository_list}; do
+                        first_part=$(echo "$repo_name" | cut -d'-' -f1 | tr '[:upper:]' '[:lower:]')
+                        echo "Adding repository $repo_name under $first_part"
+                        poetry source add --priority=supplemental ${first_part} https://${{ secrets.JFROG_ARTIFACTORY_URL }}/artifactory/api/pypi/${repo_name}/simple
+                        poetry config -- http-basic.${first_part} github ${{ secrets.JFROG_ARTIFACTORY_TOKEN }}
+                    done
```

Comment:
can this be a reusable action? Seems duplicated with code in `actions/reusable-python-setup_poetry/action.yml`

Reply:
acceptable, only duplicated once 😉